File: data/java/src/function_peewee_get.java
package com.example;

import org.jdbi.v3.core.Jdbi;
import org.jdbi.v3.core.Handle;
import java.util.Optional;

public class PeeweeExecutor {
    // Non-static method implementing all functionality.
    public String get_name_by_id(String db_path, int record_id) {
        String jdbcUrl = "jdbc:sqlite:" + db_path;
        // Create a Jdbi instance using the SQLite JDBC driver.
        Jdbi jdbi = Jdbi.create(jdbcUrl);
        try (Handle handle = jdbi.open()) {
            // Drop the table if it exists to ensure a fresh state for tests.
            handle.execute("DROP TABLE IF EXISTS TestModel");
            // Create table.
            handle.execute("CREATE TABLE TestModel (id INTEGER PRIMARY KEY, name TEXT)");

            // Insert test data.
            handle.execute("INSERT INTO TestModel (id, name) VALUES (?, ?)", 1, "Alice");
            handle.execute("INSERT INTO TestModel (id, name) VALUES (?, ?)", 2, "Bob");

            // Query the name for the given record_id.
            Optional<String> result = handle.createQuery("SELECT name FROM TestModel WHERE id = ?")
                    .bind(0, record_id)
                    .mapTo(String.class)
                    .findOne();

            return result.orElse("Not Found");
        } catch (Exception e) {
            return "Not Found";
        }
    }
}

File: data/python/src/function_peewee_get.py
from peewee import SqliteDatabase, Model, CharField, IntegerField

class PeeweeExecutor:
    def get_name_by_id(self, db_path: str, record_id: int) -> str:
        db = SqliteDatabase(db_path)

        class BaseModel(Model):
            class Meta:
                database = db

        class TestModel(BaseModel):
            id = IntegerField(primary_key=True)
            name = CharField()

        db.connect()
        db.create_tables([TestModel], safe=True)

        # 插入测试数据
        if not TestModel.select().where(TestModel.id == 1).exists():
            TestModel.create(id=1, name="Alice")
            TestModel.create(id=2, name="Bob")

        # 查询指定 ID 的 name 值
        try:
            result = TestModel.get(TestModel.id == record_id)
            return result.name
        except TestModel.DoesNotExist:
            return "Not Found"
        finally:
            db.close()
